diff --git a/src/android/app/build.gradle.kts b/src/android/app/build.gradle.kts
index 87d815c92c..ad3688d13e 100644
--- a/src/android/app/build.gradle.kts
+++ b/src/android/app/build.gradle.kts
@@ -58,7 +58,7 @@ android {
     }
 
     defaultConfig {
-        applicationId = "dev.eden.eden_emulator"
+        applicationId = "dev.eden.eden_nightly"
         minSdk = 24
         targetSdk = 36
         versionName = getGitVersion()
@@ -158,7 +158,13 @@ android {
     productFlavors {
         create("mainline") {
             dimension = "version"
-            resValue("string", "app_name_suffixed", "Eden")
+            resValue("string", "app_name_suffixed", "Eden Nightly")
+
+            sourceSets {
+                getByName("mainline") {
+                    res.srcDirs("src/main/legacy")
+                }
+            }
 
             ndk {
                 abiFilters += listOf("arm64-v8a")
